### PR TITLE
Look for `setImmediate` in all possible locations.

### LINKED ISSUE #25: Uncaught ReferenceError: setImmediate is not defined
When use with webpack, there maybe an Exception thrown out under some condition.

### PR DESCRIPTION
**Pull request by darionco**
Fixes #25
Looks for setImmediate in all objects in the same order as the `setimmediate` library adds them, if not found, it falls back to using `setTimeout` and `clearTimeout` which behave very similarly to `setImmediate` when no parameters passed.

### CODE REVIEW

**Comment by jryans** [comment]
Out of curiosity, what's your setup like where this is necessary? Just like with #25, I am curious why I only see a few sporadic reports of this.

**Comment by darionco** [comment]
My use case is somewhat peculiar. I am working on creating a visualization for PowerBI (https://powerbi.microsoft.com/en-us/) they sandbox the visuals in an iframe, when the code is loaded inside the iframe they freeze the window object for security reasons. To allow for other libraries to load correctly they do provide a `self` object (which seems to be a proxy of window[?!]) but will not write to the actual `window` object... it's weird.

In short, the `setimmediate` library is loaded in the `self` object, which is not the `window` object so accessing the methods via `setImmediate`/`clearImmediate` throws an error while accessing them via `self.setImmediate`/`self.clearImmediate` works just fine.

File: main.js [hunk 1]
```diff
@@ -49,5 +49,5 @@ exports._unrefActive = exports.active = function(item) {
 
 // setimmediate attaches itself to the global object
 require("setimmediate");
-exports.setImmediate = setImmediate;
-exports.clearImmediate = clearImmediate;
+exports.setImmediate = self && self.setImmediate || global && global.setImmediate || window && window.setImmediate || setInterval;
```
Comment (jryans) [inline main.js]: Please add a comment in code about this being the order that `setimmediate` tries to attach to the global. Something about this being occasionally necessary in exotic environments might be nice as a comment as well.

Also, it looks like the library tries: `self`, `global`, and `this`. To be more accurate, shouldn't `window` be replaced with `this` instead?

Comment (darionco) [reply main.js]: I have addressed the comments and updated the PR, thinking about it though, `require` is a function (and webpack wraps `require`s in its own functions) wouldn't that mean that `this` would be the require function itself?

Comment (jryans) [reply main.js]: Hmm... I would hope Webpack is okay with this, as I feel like many other things would break if it was not.

I tried a super-quick test with Webpack, and it appears to work with you latest patch, so let's proceed.

**Comment by jryans** [comment]
Thanks for working on this!

**Comment by jryans** [comment]
Published as v2.0.5.
